### ISSUE TITLE
Gradle Plugin: Support isolated projects

### ISSUE THREAD

**Issue by hfhbd** [google/ksp#1752]
- Replace `findProperty` with `providers.gradleProperty`
- Ideally, KGP adds more provider (and DomainCollection) APIs instead Observable/Lazy, ConfigurationAwareProvider...

**Comment by hfhbd** [comment]
Can you please reopen this issue? I still get two issues when using 2.0.20-1.0.25-SNAPSHOT:

- [ ] https://github.com/google/ksp/blob/429f173a3f77836a70a41d633c0a680a7c55aa5e/gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt#L901
- [ ] https://github.com/google/ksp/blob/429f173a3f77836a70a41d633c0a680a7c55aa5e/gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt#L349

Basically, you can't fetch the tasks list from the dependencies: `org.gradle.api.InvalidUserCodeException: Project ':foo' cannot access task dependencies directly` if you use a multi module setup `:foo` that depends on `:bar` using `implementation` given a Kotlin/JVM project.

**Comment by ganadist** [comment]
I tried to enable Isolated Project with KSP 2.0.2
Then, gradle shows warnings at following locations.
* https://github.com/google/ksp/blob/b1d4a98418cec00aac3e964c21ec114ff63a0b8a/gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspAATask.kt#L194
  ```
  cannot access task dependencies directly
    - Exception at `com.google.devtools.ksp.gradle.KspAATask$Companion$registerKspAATask$kspTaskProvider$1.invoke(KspAATask.kt:194)`
  ```
* https://github.com/google/ksp/blob/b1d4a98418cec00aac3e964c21ec114ff63a0b8a/gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt#L387
  ```
  cannot access task dependencies directly
     - Exception at `com.google.devtools.ksp.gradle.KspGradleSubplugin.applyToCompilation$configureAsAbstractKotlinCompileTool(KspSubplugin.kt:387)`
  ``` 
* https://github.com/google/ksp/blob/b1d4a98418cec00aac3e964c21ec114ff63a0b8a/gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt#L934
  ```
  cannot access task dependencies directly
    - Exception at `com.google.devtools.ksp.gradle.KspSubpluginKt.nonSelfDeps(KspSubplugin.kt:934)`
  ```

**Comment by mpartel-oura** [comment]
Did https://github.com/google/ksp/pull/2713 fix this? If so, would it be possible to do a release soon?

**Comment by solrudev** [comment]
I've tested the recently released [2.3.4](https://github.com/google/ksp/releases/tag/2.3.4) version containing #2713 together with Kotlin 2.3.0, and project isolation seems to be working without problems.

**Comment by hfhbd** [comment]
Yeah, I also tested it with a K/JVM project.
